specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json

```diff
@@ -477,6 +537,233 @@
     }
   },
   "definitions": {
+    "AppServiceEnvironment": {
+      "description": "Description of an App Service Environment.",
+      "required": [
+        "name",
+        "location",
+        "virtualNetwork",
+        "workerPools"
+      ],
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Name of the App Service Environment.",
+          "type": "string"
+        },
+        "location": {
+          "description": "Location of the App Service Environment, e.g. \"West US\".",
+          "type": "string"
+        },
+        "provisioningState": {
+          "description": "Provisioning state of the App Service Environment.",
+          "enum": [
+            "Succeeded",
+            "Failed",
+            "Canceled",
+            "InProgress",
+            "Deleting"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": false
+          }
+        },
+        "status": {
+          "description": "Current status of the App Service Environment.",
+          "enum": [
+            "Preparing",
+            "Ready",
+            "Scaling",
+            "Deleting"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "HostingEnvironmentStatus",
+            "modelAsString": false
+          }
+        },
+        "vnetName": {
```

Comment:
AppServiceEnvironment is not an ARM Resource. The model 'AppServiceEnvironmentResource' is the ARM Resource.

Reply:
w.r.t PATCH change. 
1. We now have the x-ms-mutability extension on there. Shouldn't that guarantee that location is required for constructor and is read-only otherwise? 
2. Specifically which new properties? We normally change api-version only if there is a breaking change in contract or API behavior. 
3. The patch APIs actually accept the same payloads as PUT since they are internally the same REST API methods. I will have to check if all properties in the payload are patch-able.